### PR TITLE
Pflickin/refresh

### PR DESCRIPTION
**Pull request by pflickin**
- Updated the Python logging to use the new Opencensus azure packages.
- Removed uwsgi in favor of gunicorn.
- Updated the R base image to use a supported GDAL base.
- Several modifications to the ai4e_service and examples.
- Change python images to use guid task ids.
- Update R example and update SAS library for latest Azure Storage library.

### CODE REVIEW

File: Containers/base-py/Dockerfile [hunk 1]
```diff
@@ -2,24 +2,48 @@
...
+RUN easy_install3 pip
+
+# Install hosting requirements
+RUN pip3 install flask \
+        flask-restful \
+        gunicorn
 
 # Install Azure Blob SDK
-RUN ./ai4e_api_tools/requirements/install-azure-blob.sh
+RUN pip3 install azure-mgmt-storage \
```
Comment (yangsiyu007) [inline Containers/base-py/Dockerfile:39]: Should we pin the versions of the Azure packages or record them somewhere? Each time the base images are built, these versions may change.

File: Examples/base-py/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,15 @@
-FROM mcr.microsoft.com/aiforearth/base-py:1.5
-
-# Example of installing additonal Anaconda packages - numpy and pandas
-RUN echo "source activate ai4e_py_api" >> ~/.bashrc \
-    && conda install -c conda-forge -n ai4e_py_api numpy pandas
```
Comment (yangsiyu007) [inline Examples/base-py/Dockerfile:5]: Actually one more question - it'd be good to show how to install additional pip and conda packages. Is conda still available?

File: Containers/base-py/ai4e_api_tools/ai4e_app_insights.py [hunk 3]
```diff
@@ -102,28 +100,16 @@ def init_app(self, app, context):
...
-            self._key_grantee = None
-        
-        if (self._key_ai4e and len(self._key_ai4e.strip()) > 0):
-            self._key_ai4e = self._key_ai4e.strip()
-        else:
-            self._key_ai4e = None
+            self._appinsights_key = None
 
-        if self._key_grantee:
+        if self._appinsights_key:
```
[the remark this answers is not in the document]
Comment (pflickin) [reply Containers/base-py/ai4e_api_tools/ai4e_app_insights.py:110]: This is the same app insights key that any user would use in production.  Comment added and changed output message.

File: Containers/base-py/ai4e_api_tools/task_management/api_task.py [hunk 4]
```diff
@@ -44,10 +78,16 @@ def FailTask(self, taskId, status):
...
+                statuses = json.load(f)
+
+                for rec_status in statuses:
+                    if (rec_status['TaskId'] == taskId):
+                        return rec_status
+        status = {}
+        status['TaskId'] = taskId
+        status['Status'] = 'created'
+        status['Timestamp'] = datetime.datetime.strftime(datetime.datetime.now(), "%Y-%m-%d %H:%M:%S")
+        status['Endpoint'] = ''
```
[the remark this answers is not in the document]
Comment (pflickin) [reply Containers/base-py/ai4e_api_tools/task_management/api_task.py:92]: The endpoint corresponds to the endpoint that is being tracked by the taskId.  It is only empty if there is an internal error.  I've changed the status from 'created' to 'Not found.'

File: Examples/base-py/Dockerfile (hunk 2 above)
Comment (pflickin) [reply Examples/base-py/Dockerfile:5]: The API bases have been converted to use pip for all installs, so Conda is not needed.  I added a pip install example in this API.

File: Containers/base-py/Dockerfile [hunk 5]
```diff
@@ -2,24 +2,48 @@
...
+
+RUN easy_install3 pip
+
+# Install hosting requirements
+RUN pip3 install flask \
+        flask-restful \
+        gunicorn
 
 # Install Azure Blob SDK
-RUN ./ai4e_api_tools/requirements/install-azure-blob.sh
```
[the remark this answers is not in the document]
Comment (pflickin) [reply Containers/base-py/Dockerfile:14]: Created a versioned requirements.txt pip file for each base image

**Comment by pflickin** [comment]
- Updated the Python logging to use the new Opencensus azure packages.
- Removed uwsgi in favor of gunicorn.
- Updated the R base image to use a supported GDAL base.
- Change python images to use guid task ids.
- Several modifications to the ai4e_service and examples.
- Image version updates and documentation.
- Update R example and update SAS library for latest Azure Storage library.
- Removed separate app insights key requirement.
- Added versions to all pip packages within requirements.txt files.
- Resolved issues in the task manager (missing task, utc time, etc.)

**Pull request merged**